design/ErrorHandling.md

```diff
@@ -80,90 +79,244 @@ device.addEventListener('validationerror', (event) => {
 });
 ```
 
-## Object Creation
+## Object and Operation Status
+
+### Internal Nullability
 
 WebGPU objects are opaque handles.
 On creation, such a handle is "pending" until the backing object is created by the implementation.
 After that, a handle may refer to a successfully created object (called a "valid object"), or an error that occured during creation (called an "invalid object").
 
-When a WebGPU object handle is passed to an operation, the object will resolve (to "valid" or "invalid") before it is actually used by that operation.
+When a WebGPU object handle is passed to a WebGPU API call, the object must resolve (to "valid" or "invalid") before the implementation can perform validation on that call.
```

Comment:
nit: for someone unfamiliar with our error monad, this would sound like it would resolve in the Javascript task so that after a call you can ask synchronously if the object is valid.